core/src/main/java/io/grpc/transport/AbstractClientStream.java

```diff
@@ -150,14 +155,14 @@ protected void inboundDataReceived(ReadableBuffer frame) {
   }
 
   @Override
-  protected void inboundDeliveryPaused() {
+  protected final void inboundDeliveryPaused() {
     runCloseListenerTask();
   }
 
   @Override
   protected final void deframeFailed(Throwable cause) {
     log.log(Level.WARNING, "Exception processing message", cause);
-    cancel(Status.CANCELLED);
+    cancel(Status.INTERNAL);
```

Comment:
Hmm... actually, it looks like maybe we should remove the log and do:
cancel(Status.INTERNAL.withDescription("Exception processing message").withCause(cause));

This line is quite old and predates us having a clear idea on processing exceptions. Maybe leave it as you have (with INTERNAL), but have a TODO to investigate what is appropriate.